src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotIgnoreReturnValueCSharpTests.cs

```diff
@@ -0,0 +1,163 @@
+        [Fact]
+        public async Task UnannotatedMethod_DiscardingReturnValue_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync($$"""
+                {{attributeImplementationCSharp}}
+
+                class C
+                {
+                    int UnannotatedMethod() => 1;
+
+                    void M()
+                    {
+                        _ = UnannotatedMethod();
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public async Task AnnotatedMethod_IgnoringReturnValue_ProducesDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync($$"""
+                {{attributeImplementationCSharp}}
+
+                class C
+                {
+                    [return: System.Diagnostics.CodeAnalysis.DoNotIgnore]
+                    int AnnotatedMethod() => 1;
+
+                    void M()
+                    {
+                        {|#1:AnnotatedMethod()|};
+                    }
+                }
+                """,
+                VerifyCS.Diagnostic(doNotIgnoreRule).WithLocation(1).WithArguments("C.AnnotatedMethod()")
+            );
+        }
+
+        [Fact]
+        public async Task AnnotatedMethod_IgnoringReturnValue_ProducesDiagnostic_WithMessage()
+        {
+            await VerifyCS.VerifyAnalyzerAsync($$"""
+                {{attributeImplementationCSharp}}
+
+                class C
+                {
+                    [return: System.Diagnostics.CodeAnalysis.DoNotIgnore(Message = "You need this 1")]
+                    int AnnotatedMethod() => 1;
+
+                    void M()
+                    {
+                        {|#1:AnnotatedMethod()|};
+                    }
+                }
+                """,
+                VerifyCS.Diagnostic(doNotIgnoreRuleWithMessage).WithLocation(1).WithArguments("C.AnnotatedMethod()", "You need this 1")
+            );
+        }
+
+        [Fact]
+        public async Task AnnotatedMethod_ConsumingReturnValue_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync($$"""
+                {{attributeImplementationCSharp}}
+
+                class C
+                {
+                    [return: System.Diagnostics.CodeAnalysis.DoNotIgnore]
+                    int AnnotatedMethod() => 1;
+
+                    void M()
+                    {
+                        int r = AnnotatedMethod();
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public async Task AnnotatedMethod_DiscardingReturnValue_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync($$"""
+                {{attributeImplementationCSharp}}
+
+                class C
+                {
+                    [return: System.Diagnostics.CodeAnalysis.DoNotIgnore]
+                    int AnnotatedMethod() => 1;
+
+                    void M()
+                    {
+                        _ = AnnotatedMethod();
```

Comment:
It'd be good to have tests that check other usage, e.g. `CallSomethingElse(AnnotatedMethod());`, as the expression used in a switch expression case, as one of the expressions of a ternary, etc.